### PR TITLE
fix(tds-dropdown): improved scrollbar styling

### PR DESCRIPTION
**Pull request by finnlake**
**Describe pull-request**  
_Describe what the pull-request is about_

**Solving issue**  
_Please link to the issue this pull request is solving by adding ticket number after CDEP-_
Fixes: [CDEP-3205](https://tegel.atlassian.net/browse/CDEP-3205)

**How to test**  
_Provide detailed steps for testing, including any necessary setup._
1. Test dropdown in storybook
2. More options added in the story so scrollbar should be visible

**Checklist before submission**
- [ ] I have added unit tests for my changes (if applicable)
- [x ] All existing tests pass
- [ ] I have updated the documentation (if applicable)

**Suggested test steps**
- [ x] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [x ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

**Screenshots**  
_Include before/after screenshots for UI changes._

**Additional context**  
_Add any other context or feedback requests about the pull-request here._

[CDEP-3205]: https://tegel.atlassian.net/browse/CDEP-3205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/components/dropdown/dropdown.scss [hunk 1]
```diff
@@ -131,7 +173,17 @@
     box-shadow: rgb(0 0 0 / 10%) 0 2px 3px 0;
     border-radius: 0 0 4px 4px;
     overflow-y: auto;
-    @include tds-scrollbar;
+    @include tds-dropdown-scrollbar;
+
+    // TODO: Why cant we put this block in the mixin tds-dropdown-scrollbar? Can not get that to work.
```
Comment (timrombergjakobsson) [inline packages/core/src/components/dropdown/dropdown.scss:178]: Regarding this TODO, a mixin itself doesn't have any inherent logic capabilities to detect browser support. It blindly outputs the styles wherever it's included. So, if a mixin includes both -webkit- prefixed styles for scrollbar customization and @supports not rules intended for browsers that do not support those prefixes, it cannot dynamically adjust based on the client's browser.

File: packages/core/src/components/dropdown/dropdown-vars.scss [hunk 2]
```diff
@@ -10,6 +10,9 @@
   --tds-dropdown-helper-color: var(--tds-grey-700);
   --tds-dropdown-placeholder-color: var(--tds-grey-700);
   --tds-dropdown-disabled-color: var(--tds-grey-400);
+  --tds-dropdown-scrollbar-background: var(--tds-grey-100);
+  --tds-dropdown-scrollbar-thumb-color: var(--tds-grey-500);
```
Comment (timrombergjakobsson) [inline packages/core/src/components/dropdown/dropdown-vars.scss:14]: This should be:
`--your-css-variable-name: rgba(0, 0, 0 / 38%);`
And no hover color is needed, merely the thickening of the scrollbar when hovering.

File: packages/core/src/components/dropdown/dropdown-vars.scss [hunk 3]
```diff
@@ -10,6 +10,9 @@
   --tds-dropdown-helper-color: var(--tds-grey-700);
   --tds-dropdown-placeholder-color: var(--tds-grey-700);
   --tds-dropdown-disabled-color: var(--tds-grey-400);
+  --tds-dropdown-scrollbar-background: var(--tds-grey-100);
```
Comment (timrombergjakobsson) [inline packages/core/src/components/dropdown/dropdown-vars.scss:13]: This is not used?

File: packages/core/src/components/dropdown/dropdown-vars.scss (hunk 2 above)
Comment (finnlake) [reply packages/core/src/components/dropdown/dropdown-vars.scss:14]: Fixed

File: packages/core/src/components/dropdown/dropdown-vars.scss (hunk 3 above)
Comment (finnlake) [reply packages/core/src/components/dropdown/dropdown-vars.scss]: Refactoring error, I changed variable name from --tds-dropdown-scrollbar-background to --tds-dropdown-scrollbar-track-color. Fix committed.

**Comment by theJohnnyMe** [comment]
To help our colleague designers evaluate, here is a comparison image: 

Screenshot 2024-03-14 at 16 33 12-tile

**Comment by finnlake** [comment]
For some reason the darkmode is not behaving the same way when I test it locally and in the deployed branch.
Locally it looks good, but not in the deployed branch. Can it be that the deployed branch hasnt been updated properly...?
image

**Comment by finnlake** [comment]
Yes, a bit of SBS... :)
Now the final(!?) fix is committed.
Dark mode looks like this now

image

**Comment by lau-vt** [comment]
@finnlake thanks for your submission. I've reviewed your request and have a few minor requests for changes before we can go ahead and approve it from a design standpoint it into tegel.

1. The "rail" for the scrollbar should be transparent and not visible and overlay elements behind it.
2. Make the dark-mode scrollbar grey-700
3. Make the light-mode scrollbar grey-400

If we can align on the changes above we'll update it in the design as well since this is a change also for the design. But in that way the scrollbar provides a better contrast and usability, while also keeping consistent with the visual expression and aesthetic throughout the system

I'll add an image below for reference
Screenshot 2024-03-18 at 10 57 49

**Comment by finnlake** [comment]
> @finnlake thanks for your submission. I've reviewed your request and have a few minor requests for changes before we can go ahead and approve it from a design standpoint it into tegel.
> 
> 1. The "rail" for the scrollbar should be transparent and not visible and overlay elements behind it.
> 2. Make the dark-mode scrollbar grey-700
> 3. Make the light-mode scrollbar grey-400
> 
> If we can align on the changes above we'll update it in the design as well since this is a change also for the design. But in that way the scrollbar provides a better contrast and usability, while also keeping consistent with the visual expression and aesthetic throughout the system
> 
> I'll add an image below for reference Screenshot 2024-03-18 at 10 57 49

1. It dont think that is possible with the webkit scrollbar styling that we are using. The webkit scrollbar is a classic scrollbar and not an overlay scrollbar see https://developer.chrome.com/docs/css-ui/scrollbar-styling.
Making the scrollbar track transparent is possible, but the options will still end before the track area. Because of that I think we should keep the scrollbar as a classic scrollbar.

**Comment by mJarsater** [comment]
> > @finnlake thanks for your submission. I've reviewed your request and have a few minor requests for changes before we can go ahead and approve it from a design standpoint it into tegel.
> > 
> > 1. The "rail" for the scrollbar should be transparent and not visible and overlay elements behind it.
> > 2. Make the dark-mode scrollbar grey-700
> > 3. Make the light-mode scrollbar grey-400
> > 
> > If we can align on the changes above we'll update it in the design as well since this is a change also for the design. But in that way the scrollbar provides a better contrast and usability, while also keeping consistent with the visual expression and aesthetic throughout the system
> > I'll add an image below for reference Screenshot 2024-03-18 at 10 57 49
> 
> 1. It dont think that is possible with the webkit scrollbar styling that we are using. The webkit scrollbar is a classic scrollbar and not an overlay scrollbar see https://developer.chrome.com/docs/css-ui/scrollbar-styling.
>    Making the scrollbar track transparent is possible, but the options will still end before the track area. Because of that I think we should keep the scrollbar as a classic scrollbar.

I don't know what the timeline is for this change. But if it's not a suuuper urgent issue it could be worth looking into https://kingsora.github.io/OverlayScrollbars/

It's an overlaysscrollbars lib that can be used to create the style design is requesting. I'd also recomment the team to expose some sort of `tds-scroll-area` component that users could wrap their scrollable content in, to get a consistency on the scrollbar look and feel.

**Review by timrombergjakobsson — APPROVED** [review]
Given the current limitations with `::-webkit`, this is as good as we can get it right now. To make it 💯  aligned with Figma we need a different approach, something to iterate on going forward.

File: packages/core/src/components/dropdown/dropdown.scss [hunk 4]
```diff
@@ -1,8 +1,10 @@
 @import '../../mixins/z-index';
-@import '../../mixins/scrollbar';
 @import './dropdown-button';
 @import './dropdown-filter';
 
+// TODO: Use specific scrollbar styling to begin with instead of common mixin
```
Comment (timrombergjakobsson) [inline packages/core/src/components/dropdown/dropdown.scss:5]:
```suggestion
```

**Review by theJohnnyMe — APPROVED** [review]
Thanks a lot @timrombergjakobsson for resolving this one.
Looks good to me.

**Pull request merged**